gravitee-apim-console-webui/src/management/api/portal/subscriptions/components/dialogs/creation/api-portal-subscription-creation-dialog.component.spec.ts

```diff
@@ -207,6 +214,198 @@ describe('Subscription creation dialog', () => {
+                });
+                return constants;
+              },
+            },
+          ],
+        });
+        fixture = TestBed.createComponent(TestComponent);
+        httpTestingController = TestBed.inject(HttpTestingController);
+        fixture.detectChanges();
+        component = fixture.componentInstance;
+        loader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+      });
+
+      afterEach(() => {
+        jest.clearAllMocks();
+        httpTestingController.verify();
+      });
+
+      it('should not be able to select apikey mode when no subscription and custom api key', async () => {
+        const applicationWithClientId = fakeApplication({
+          id: 'my-app',
+          name: 'withClientId',
+          settings: { app: { client_id: 'clientId' } },
+          api_key_mode: ApiKeyMode.UNSPECIFIED,
+        });
+        const planV4 = fakePlanV4({ apiId: 'my-api', mode: 'STANDARD', security: { type: 'API_KEY' }, generalConditions: undefined });
+        component.plans = [planV4];
+        component.availableSubscriptionEntrypoints = [];
+
+        jest.spyOn(TestBed.inject(SubscriptionService), 'getApplicationSubscriptions').mockReturnValue(
+          of(<PagedResult>{
+            data: [
+              {
+                security: PlanSecurityType.API_KEY,
+                api: {
+                  id: 'another-plan-id',
+                },
+              },
+            ],
+          }),
+        );
+
+        await componentTestingOpenDialog();
+
+        const harness = await loader.getHarness(ApiPortalSubscriptionCreationDialogHarness);
+        expect(await harness.isPlanRadioGroupDisabled()).toBeTruthy();
+
+        expect(await harness.isCustomApiKeyInputDisplayed()).toBeFalsy();
+        expect(await harness.isApiKeyModeRadioGroupDisplayed()).toBeFalsy();
+
+        await harness.searchApplication('withClientId');
+        expectApplicationsSearch('withClientId', [applicationWithClientId]);
+        await harness.selectApplication(applicationWithClientId.name);
+
+        expect(await harness.isPlanRadioGroupDisabled()).toBeFalsy();
+        await harness.choosePlan(planV4.name);
+
+        expect(await harness.isApiKeyModeRadioGroupDisplayed()).toBeTruthy();
+
+        await harness.chooseApiKeyMode('API Key');
+
+        expect(await harness.isCustomApiKeyInputDisplayed()).toBeTruthy();
+
+        await harness.addCustomKey('12345678');
+        const req = httpTestingController.expectOne({
+          url: `${CONSTANTS_TESTING.env.v2BaseURL}/apis/my-api/subscriptions/_verify`,
+          method: 'POST',
+        });
+        const verifySubscription: VerifySubscription = {
+          applicationId: 'my-app',
+          apiKey: '12345678',
+        };
+        expect(req.request.body).toEqual(verifySubscription);
+        req.flush({ ok: true });
+      });
+      it('should be able to select api key mode EXCLUSIVE and custom api key', async () => {
```

Comment:
the title does not fit the content of the test.

Reply:
good catch